client/src/components/Flex.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { ReactNode } from 'react';
+
+type FlexProps = {
+  children?: ReactNode;
+  align?: string;
+  justify?: string;
+  direction?: string;
+  flexWrap?: string;
+};
+
+const Flex: React.FC<FlexProps> = ({
+  children,
+  direction,
+  flexWrap,
+  justify,
+  align,
+  ...props
+}: FlexProps) => {
+  if ('row' === direction) {
+    return <div className="flex flex-row ...">{children}</div>;
+  } else if ('column' === direction) {
```

Comment:
You can create a variable that carry the classes name `classes`, and let it return after reading the value of the `classes` instead for readiblity

Here is the logic.

```
Create classes with defaultClass

If variable exist
Add variable to classes

Render div className={classes}
```

Reply:
Got it! That made total sense. I'll try to implement it and my migrate all my code to my main branch. Besides, do you think there is anything to modify for the business logic for Flex component.

Reply:
@Sboonny On the side note, I think it would be less error-prone if you have the set up to the main codebase, and then I can test my code since I'm not totally sure if there would be any conflicts down the road if I merge my main branch to the Chapter's code base right away. Once you're done with setting up I will pull all the changes for the set up and then I can merge my components and storybooks to it. Let me know what you think.

Reply:
> if you have the set up to the main codebase

What do you have in your mind, that can help with this issue?

> I can test my code since I'm not totally sure if there would be any conflicts down the road if I merge my main branch to the Chapter's code base right away. Once you're done with setting up I will pull all the changes for the set up and then I can merge my components and storybooks to it

If you are talking about script that run something like this

```ts
describe('<Component />', () => {
  it('should render correctly', () => {
    render(
      <Component></Component>
    );
  }),
```

Reply:
I'm worried about Tailwindcss configuration and package.json since I saw a lot of conflicts when I tried to merge my UI branch to my updated main branch.

Reply:
You don't need to have tailwind in chapter, I helped you install it, because I thought you wanted to get the feeling for them.

Reply:
The ui components pull request should be opened in freeCodeCamp and merge into freeCodeCamp ui components issue

Reply:
If your assignment forces you into creating pull request into chapters, I can create other issues and assign you and the others to it.

Reply:
> The ui components pull request should be opened in freeCodeCamp and merge into freeCodeCamp ui components issue

Is it okay if I fix my code and use plain css and settle this Flex issue? I feel that I already made progress and I want to get it done. Then I plan to work in the issue #1244 after I have my custom Flex component.

Reply:
I think it's done, tailwind is adding flex class like wrap, no-wrap and similar. Adding more will reduce the scenarios in which the component is used.

just make the logic readible and open it in freeCodeCamp repo.